src/components/GroupList/GroupList.js

```diff
@@ -0,0 +1,33 @@
+/* @flow */
+
+import React, { PropTypes } from 'react'
+import _ from 'lodash'
+import { List } from 'material-ui/List'
+import GroupItem from './GroupItem'
+
+function GroupList(props) {
+  const { data } = props
+  return (
+    <List
+      style={{ paddingTop: 0, paddingBottom: 0 }}
+      primaryTogglesNestedList
+    >
+      {data && data.length &&
+        data.map(item =>
+          <GroupItem
+            key={_.uniqueId('_group_item')}
```

Comment:
so generating with the lodash is correct as long as we looping here, incorrect that we create new ID every time parent component renders, which is correct if items changed or incorrect if they are not - but they got new unigue keys :) So we either should be careful with their uniqueness(cache them), or create proper `shouldComponentUpdate`  to support proper reconciliation.